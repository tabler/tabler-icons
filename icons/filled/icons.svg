<!--
unicode: "10070"
version: "3.19"
-->
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="currentColor"
>
    <path d="M4.25 2.603a4.5 4.5 0 1 1 -2.25 3.897l.006 -.229a4.5 4.5 0 0 1 2.244 -3.668M5.632 13.504a1 1 0 0 1 1.736 0l4 7a1 1 0 0 1 -.868 1.496h-8a1 1 0 0 1 -.868 -1.496zM13.293 2.293a1 1 0 0 1 1.414 0l7 7a1 1 0 1 1 -1.414 1.414l-7 -7a1 1 0 0 1 0 -1.414M20.293 2.293a1 1 0 0 1 1.414 1.414l-7 7a1 1 0 0 1 -1.414 -1.414zM21 13a1 1 0 0 1 1 1v7a1 1 0 0 1 -1 1h-7a1 1 0 0 1 -1 -1v-7a1 1 0 0 1 1 -1z" />
</svg>
