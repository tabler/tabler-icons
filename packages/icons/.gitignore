tabler-nodes.json
tabler-nodes-filled.json
tabler-nodes-outline.json
tabler-sprite-nostroke.svg
tabler-sprite.svg
tags.json
icons.json
categories/*
