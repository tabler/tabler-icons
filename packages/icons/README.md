# Tabler Icons

<p align="center">
  <img src="https://raw.githubusercontent.com/tabler/tabler-icons/master/.github/packages/og-core.png" alt="Tabler Icons" width="838">
</p>

<p align="center">
    A set of <!--icons-count-->4694<!--/icons-count--> free MIT-licensed high-quality SVG icons for you to use in your web projects. Each icon is designed on a 24x24 grid and a 2px stroke.
<p>

<p align="center">
  <a href="https://tabler-icons.io/"><strong>Browse all icons at tabler-icons.io &rarr;</strong></a>
</p>

<p align="center">
    <a href="https://github.com/tabler/tabler-icons/releases"><img src="https://img.shields.io/npm/v/@tabler/icons" alt="Latest Release"></a>
    <a href="https://github.com/tabler/tabler-icons/blob/master/LICENSE"><img src="https://img.shields.io/npm/l/@tabler/icons.svg" alt="License"></a>
</p>

## Sponsors

**If you want to support my project and help me grow it, you can [become a sponsor on GitHub](https://github.com/sponsors/codecalm) or just [donate on PayPal](https://paypal.me/codecalm) :)**

<a href="https://github.com/sponsors/codecalm">
  <img src='https://raw.githubusercontent.com/tabler/static/main/sponsors.png'>
</a>

## Installation

```
yarn add @tabler/icons
```

or

```
npm install @tabler/icons
```

or

```
pnpm install @tabler/icons
```

or just [download from Github](https://github.com/tabler/tabler-icons/releases).

## Usage

All icons are built with SVG, so you can place them as `<img>`, `background-image` and inline in HTML code.

### HTML image

If you load an icon as an image, you can modify its size using CSS.

```html
<img src="path/to/icon.svg" alt="icon title" />
```

### Inline HTML

You can paste the content of the icon file into your HTML code to display it on the page.

```html
<a href="">
  <svg
    xmlns="http://www.w3.org/2000/svg"
    class="icon icon-tabler icon-tabler-disabled"
    width="24"
    height="24"
    viewBox="0 0 24 24"
    stroke-width="1.25"
    stroke="currentColor"
    fill="none"
    stroke-linecap="round"
    stroke-linejoin="round"
  >
    ...
  </svg>
  Click me
</a>
```

Thanks to that, you can change the size, color and the `stroke-width` of the icons with CSS code.

```css
.icon-tabler {
  color: red;
  width: 32px;
  height: 32px;
  stroke-width: 1.25;
}
```

### SVG sprite

Add an icon to be displayed on your page with the following markup (`activity` in the above example can be replaced with any valid icon name):

```html
<svg width="24" height="24">
  <use xlink:href="path/to/tabler-sprite.svg#tabler-activity" />
</svg>
```

## Contributing

For more info on how to contribute please see the [contribution guidelines](https://github.com/tabler/tabler-icons/blob/main/CONTRIBUTING.md).

Caught a mistake or want to contribute to the documentation? [Edit this page on Github](https://github.com/tabler/tabler-icons/blob/main/packages/icons/README.md)

## License

Tabler Icons is licensed under the [MIT License](https://github.com/tabler/tabler-icons/blob/master/LICENSE).

## Sponsor Tabler

<a href="https://github.com/sponsors/codecalm" target="_blank"><img src="https://github.com/tabler/tabler/raw/dev/src/static/sponsor-banner-readme.png?raw=true" alt="Sponsor Tabler" /></a>
