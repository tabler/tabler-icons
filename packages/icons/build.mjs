import fs from 'fs'
import { getAllIconsMerged, getAllIcons } from '../../.build/helpers.mjs'

const iconsMerged = getAllIconsMerged()
const icons = getAllIcons(true, true)

const buildJSON = () => {
  console.log('Building icons.json')

  fs.writeFileSync(`./icons.json`, JSON.stringify(iconsMerged, null, 2))
}

const buildNodes = () => {
  console.log('Building tabler-nodes.json')

  Object.entries(icons).forEach(([type, icons]) => {
    const iconNodes = icons.reduce((acc, { name, obj }) => {
      acc[name] = obj.children.map(({ name, attributes }) => [name, attributes]);

      return acc;
    }, {});

    const iconNodesStringified = JSON.stringify(iconNodes, null, 2);

    fs.writeFileSync(`./tabler-nodes-${type}.json`, iconNodesStringified);
  })
}

const buildSvgs = () => {
  console.log('Building svgs')

  Object.entries(icons).forEach(([type, icons]) => {
    fs.mkdirSync(`./icons/${type}`, { recursive: true });

    icons.forEach(({ name, content, category }) => {
      fs.writeFileSync(`./icons/${type}/${name}.svg`, content);

      if (category) {
        fs.mkdirSync(`./categories/${type}/${category}`, { recursive: true });
        fs.writeFileSync(`./categories/${type}/${category}/${name}.svg`, content);
      }
    })
  })
}

buildJSON()
buildNodes()
buildSvgs()
