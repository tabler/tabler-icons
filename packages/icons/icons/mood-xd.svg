<svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-mood-xd" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0" />
  <path d="M9 14h6a3 3 0 0 1 -6 0z" />
  <path d="M9 8l6 3" />
  <path d="M9 11l6 -3" />
</svg>


