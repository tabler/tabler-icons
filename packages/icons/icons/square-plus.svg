<svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-square-plus" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 12h6" />
  <path d="M12 9v6" />
  <path d="M3 5a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v14a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-14z" />
</svg>


