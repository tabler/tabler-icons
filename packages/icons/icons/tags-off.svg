<svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-tags-off" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M16.296 12.296l-5.71 -5.71" />
  <path d="M6 6h-1a2 2 0 0 0 -2 2v4.172a2 2 0 0 0 .586 1.414l5.71 5.71a2.41 2.41 0 0 0 3.408 0l3.278 -3.278" />
  <path d="M18 19l.496 -.496" />
  <path d="M20.384 16.367a4.822 4.822 0 0 0 -.792 -5.775l-4.592 -4.592" />
  <path d="M7 10h-.01" />
  <path d="M3 3l18 18" />
</svg>


