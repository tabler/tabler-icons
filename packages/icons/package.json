{
  "name": "@tabler/icons",
  "version": "2.35.0",
  "license": "MIT",
  "author": "codecalm",
  "description": "A set of free MIT-licensed high-quality SVG icons for you to use in your web projects.",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tabler/tabler-icons.git"
  },
  "main": "dist/cjs/tabler-icons.js",
  "main:umd": "dist/umd/tabler-icons.js",
  "exports": {
    "./*": [
      "./icons/*"
    ]
  },
  "module": "dist/esm/tabler-icons.js",
  "unpkg": "dist/umd/tabler-icons.min.js",
  "typings": "dist/tabler-icons.d.ts",
  "sideEffects": false,
  "files": [
    "dist/*",
    "tags.json",
    "tabler-nodes.json",
    "icons/*",
    "categories/*",
    "tabler-sprite.svg",
    "tabler-sprite-nostroke.svg"
  ],
  "homepage": "https://tabler-icons.io",
  "bugs": {
    "url": "https://github.com/tabler/tabler-icons/issues"
  },
  "funding": {
    "type": "github",
    "url": "https://github.com/sponsors/codecalm"
  },
  "scripts": {
    "build": "pnpm run clean && pnpm run copy && pnpm run build:icons && pnpm run build:es && pnpm run build:bundles",
    "build:icons": "node build.mjs",
    "build:es": "babel src -d dist/esm",
    "build:bundles": "rollup -c ./rollup.config.mjs",
    "copy": "pnpm run copy:license && pnpm run copy:icons && pnpm run copy:tags",
    "copy:icons": "mkdir -p ./icons && cp ../../icons/*.svg ./icons/",
    "copy:tags": "cp ../../tags.json tags.json",
    "copy:license": "cp ../../LICENSE ./LICENSE",
    "clean": "rm -rf dist && rm -rf src/icons/*.js && rm -rf icons",
    "test": "jest --env=jsdom"
  },
  "keywords": [
    "icons",
    "svg",
    "png",
    "iconfont",
    "react",
    "front-end",
    "web"
  ],
  "devDependencies": {
    "babel-jest": "^29.4.1",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1"
  },
  "jest": {
    "testEnvironmentOptions": {
      "customExportConditions": [
        "node",
        "node-addons"
      ]
    },
    "transform": {
      "^.+\\.js$": "babel-jest"
    }
  }
}
