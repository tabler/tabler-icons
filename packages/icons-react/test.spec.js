import { render } from "@testing-library/react"
import { Icon2fa } from "./src/icons.js"
import React from "react"
import renderer from 'react-test-renderer'

describe("React Icon component", () => {
  test("should render icon component", () => {
    const { container } = render(<Icon2fa/>)
    expect(container.getElementsByTagName("svg").length).toBeGreaterThan(0)
  })

  test("should update svg attributes when there are props passed to the component", () => {
    const { container } = render(<Icon2fa size={48} color={"red"} strokeWidth={4}/>)

    const svg = container.getElementsByTagName("svg")[0]

    expect(svg.getAttribute("width")).toBe("48")
    expect(svg.getAttribute("stroke")).toBe("red")
    expect(svg.getAttribute("stroke-width")).toBe("4")
  })

  // Jest creates separate file to store snapshots
  test("should match snapshot", () => {
    const icon = renderer.create(<Icon2fa/>).toJSON()
    expect(icon).toMatchSnapshot()
  })
})
