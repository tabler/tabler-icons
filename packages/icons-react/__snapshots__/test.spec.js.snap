// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Icon component should match snapshot 1`] = `
<svg
  className="tabler-icon tabler-icon-2fa"
  fill="none"
  height={24}
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth={2}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M7 16h-4l3.47 -4.66a2 2 0 1 0 -3.47 -1.54m7 6.2v-8h4m-4 4l3 0m4 4v-6a2 2 0 0 1 4 0v6m-4 -3l4 0"
  />
</svg>
`;
